From c1d6ba1310f777ff823b4c43c68f1929f9e7a199 Mon Sep 17 00:00:00 2001
From: Erik Dannenberg <erik.dannenberg@xtrade-gmbh.de>
Date: Tue, 27 Jun 2023 20:33:30 +0200
Subject: [PATCH] Revert-acct-user.eclass-Fixing-user-group-creation-w

---
 eclass/acct-user.eclass | 31 +++----------------------------
 1 file changed, 3 insertions(+), 28 deletions(-)

diff --git a/eclass/acct-user.eclass b/eclass/acct-user.eclass
index 66a4d6667888..fc631336859c 100644
--- a/eclass/acct-user.eclass
+++ b/eclass/acct-user.eclass
@@ -198,15 +198,8 @@ eislocked() {
 	*)
 		# NB: 'no password' and 'locked' are indistinguishable
 		# but we also expire the account which is more clear
-		local shadow
-		if [[ -n "${ROOT}" ]]; then
-			shadow=$(grep "^$1:" "${ROOT}/etc/shadow")
-		else
-			shadow=$(getent shadow "$1")
-		fi
-
-		[[ $( echo ${shadow} | cut -d: -f2) == '!'* ]] &&
-			[[ $(echo ${shadow} | cut -d: -f8) == 1 ]]
+		[[ $(getent shadow "$1" | cut -d: -f2) == '!'* ]] &&
+			[[ $(getent shadow "$1" | cut -d: -f8) == 1 ]]
 		;;
 	esac
 }
@@ -357,10 +350,6 @@ acct-user_pkg_preinst() {
 			opts+=( --uid "${_ACCT_USER_ID}" )
 		fi

-		if [[ -n ${ROOT} ]]; then
-			opts+=( --prefix "${ROOT}" )
-		fi
-
 		elog "Adding user ${ACCT_USER_NAME}"
 		useradd "${opts[@]}" "${ACCT_USER_NAME}" || die "useradd failed with status $?"
 		_ACCT_USER_ADDED=1
@@ -369,13 +358,7 @@ acct-user_pkg_preinst() {
 	if [[ ${_ACCT_USER_HOME} != /dev/null ]]; then
 		# default ownership to user:group
 		if [[ -z ${_ACCT_USER_HOME_OWNER} ]]; then
-			if [[ -n ${ROOT} ]]; then
-				local euid=$(egetent passwd ${ACCT_USER_NAME} | cut -d: -f3)
-				local egid=$(egetent passwd ${ACCT_USER_NAME} | cut -d: -f4)
-				_ACCT_USER_HOME_OWNER=${euid}:${egid}
-			else
-				_ACCT_USER_HOME_OWNER=${ACCT_USER_NAME}:${groups[0]}
-			fi
+			_ACCT_USER_HOME_OWNER=${ACCT_USER_NAME}:${groups[0]}
 		fi
 		# Path might be missing due to INSTALL_MASK, etc.
 		# https://bugs.gentoo.org/691478
@@ -427,10 +410,6 @@ acct-user_pkg_postinst() {
 		opts+=( --expiredate "" --unlock )
 	fi

-	if [[ -n ${ROOT} ]]; then
-		opts+=( --prefix "${ROOT}" )
-	fi
-
 	local g old_groups del_groups=""
 	old_groups=$(egetgroups "${ACCT_USER_NAME}")
 	for g in ${old_groups//,/ }; do
@@ -513,10 +492,6 @@ acct-user_pkg_prerm() {
 		--shell /sbin/nologin
 	)

-	if [[ -n ${ROOT} ]]; then
-		opts+=( --prefix "${ROOT}" )
-	fi
-
 	elog "Locking user ${ACCT_USER_NAME}"
 	usermod "${opts[@]}" "${ACCT_USER_NAME}" || die "usermod failed with status $?"
 }
--
2.41.0
