diff -rupN gentoo/eclass/acct-user.eclass gentoo-patched/eclass/acct-user.eclass
--- gentoo/eclass/acct-user.eclass	2023-01-06 13:04:25.871008804 +0100
+++ gentoo-patched/eclass/acct-user.eclass	2023-01-06 13:10:36.408013419 +0100
@@ -198,15 +198,8 @@ eislocked() {
 	*)
 		# NB: 'no password' and 'locked' are indistinguishable
 		# but we also expire the account which is more clear
-		local shadow
-		if [[ -n "${ROOT}" ]]; then
-			shadow=$(grep "^$1:" "${ROOT}/etc/shadow")
-		else
-			shadow=$(getent shadow "$1")
-		fi
-
-		[[ $( echo ${shadow} | cut -d: -f2) == '!'* ]] &&
-			[[ $(echo ${shadow} | cut -d: -f8) == 1 ]]
+		[[ $(getent shadow "$1" | cut -d: -f2) == '!'* ]] &&
+			[[ $(getent shadow "$1" | cut -d: -f8) == 1 ]]
 		;;
 	esac
 }
@@ -359,10 +352,6 @@ acct-user_pkg_preinst() {
 			opts+=( --uid "${_ACCT_USER_ID}" )
 		fi
 
-		if [[ -n ${ROOT} ]]; then
-			opts+=( --prefix "${ROOT}" )
-		fi
-
 		elog "Adding user ${ACCT_USER_NAME}"
 		useradd "${opts[@]}" "${ACCT_USER_NAME}" || die
 		_ACCT_USER_ADDED=1
@@ -371,13 +360,7 @@ acct-user_pkg_preinst() {
 	if [[ ${_ACCT_USER_HOME} != /dev/null ]]; then
 		# default ownership to user:group
 		if [[ -z ${_ACCT_USER_HOME_OWNER} ]]; then
-			if [[ -n ${ROOT} ]]; then
-				local euid=$(egetent passwd ${ACCT_USER_NAME} | cut -d: -f3)
-				local egid=$(egetent passwd ${ACCT_USER_NAME} | cut -d: -f4)
-				_ACCT_USER_HOME_OWNER=${euid}:${egid}
-			else
-				_ACCT_USER_HOME_OWNER=${ACCT_USER_NAME}:${groups[0]}
-			fi
+			_ACCT_USER_HOME_OWNER=${ACCT_USER_NAME}:${groups[0]}
 		fi
 		# Path might be missing due to INSTALL_MASK, etc.
 		# https://bugs.gentoo.org/691478
@@ -429,10 +412,6 @@ acct-user_pkg_postinst() {
 		opts+=( --expiredate "" --unlock )
 	fi
 
-	if [[ -n ${ROOT} ]]; then
-		opts+=( --prefix "${ROOT}" )
-	fi
-
 	elog "Updating user ${ACCT_USER_NAME}"
 	if ! usermod "${opts[@]}" "${ACCT_USER_NAME}" 2>"${T}/usermod-error.log"; then
 		# usermod outputs a warning if unlocking the account would result in an
@@ -481,10 +460,6 @@ acct-user_pkg_prerm() {
 		--shell /sbin/nologin
 	)
 
-	if [[ -n ${ROOT} ]]; then
-		opts+=( --prefix "${ROOT}" )
-	fi
-
 	elog "Locking user ${ACCT_USER_NAME}"
 	usermod "${opts[@]}" "${ACCT_USER_NAME}" || die
 }
