From c3e9f642680ccd0b8a8ce7c3cf5313e01ab3ac55 Mon Sep 17 00:00:00 2001
From: Erik Dannenberg <erik.dannenberg@xtrade-gmbh.de>
Date: Tue, 27 Jun 2023 20:40:12 +0200
Subject: [PATCH] Partially-revert-acct-group.eclass-inline-groupadd

---
 eclass/acct-group.eclass | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/eclass/acct-group.eclass b/eclass/acct-group.eclass
index a0ad86066309..cae7cd99f2d0 100644
--- a/eclass/acct-group.eclass
+++ b/eclass/acct-group.eclass
@@ -174,10 +174,6 @@ acct-group_pkg_preinst() {
 		opts+=( --gid "${_ACCT_GROUP_ID}" )
 	fi
 
-	if [[ -n ${ROOT} ]]; then
-		opts+=( --prefix "${ROOT}" )
-	fi
-
 	elog "Adding group ${ACCT_GROUP_NAME}"
 	groupadd "${opts[@]}" "${ACCT_GROUP_NAME}" || die "groupadd failed with status $?"
 }
-- 
2.39.3

